File: app/src/main/java/com/example/myapplication/adapters/mangaadapters/recycleradapters/RecyclerAllChapterAdapter.java
package com.example.myapplication.adapters.mangaadapters.recycleradapters;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.myapplication.R;
import com.example.myapplication.databinding.ItemListSelectChapterBinding;
import com.example.myapplication.models.mangamodels.ReadMangaModel;

import java.util.List;

public class RecyclerAllChapterAdapter extends RecyclerView.Adapter<RecyclerAllChapterAdapter.ViewHolder> {
    private Context context;
    private List<ReadMangaModel.AllChapterDatas> allChapterDatasArrayList;
    private ClickListener clickListener;

    public RecyclerAllChapterAdapter(Context context, List<ReadMangaModel.AllChapterDatas> allChapterDatasArrayList) {
        this.context = context;
        this.allChapterDatasArrayList = allChapterDatasArrayList;
        if (context instanceof ClickListener) {
            clickListener = (ClickListener) context;
        }
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(context);
        ItemListSelectChapterBinding itemListBinding = DataBindingUtil.inflate(layoutInflater, R.layout.item_list_select_chapter, parent, false);
        return new ViewHolder(itemListBinding);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        holder.itemListBinding.textViewChapterAll.setText(allChapterDatasArrayList.get(position).getChapterTitle());
        holder.itemListBinding.linearViewChapterAll.setOnClickListener(v -> clickListener.onItemClick(position, v));
    }

    @Override
    public int getItemCount() {
        return allChapterDatasArrayList.size();
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        private final ItemListSelectChapterBinding itemListBinding;

        public ViewHolder(final ItemListSelectChapterBinding itemView) {
            super(itemView.getRoot());
            this.itemListBinding = itemView;
        }
    }

    public interface ClickListener {
        void onItemClick(int position, View v);
    }
}

File: app/src/main/java/com/example/myapplication/adapters/mangaadapters/recycleradapters/RecyclerAllChapterAdapter.kt
package com.example.myapplication.adapters.mangaadapters.recycleradapters

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView

import com.example.myapplication.R
import com.example.myapplication.databinding.ItemListSelectChapterBinding
import com.example.myapplication.models.mangamodels.ReadMangaModel

class RecyclerAllChapterAdapter(private val context: Context, private val allChapterDatasArrayList: List<ReadMangaModel.AllChapterDatas>) : RecyclerView.Adapter<RecyclerAllChapterAdapter.ViewHolder>() {
    private var clickListener: ClickListener? = null

    init {
        if (context is ClickListener) {
            clickListener = context
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(context)
        val itemListBinding = DataBindingUtil.inflate<ItemListSelectChapterBinding>(layoutInflater, R.layout.item_list_select_chapter, parent, false)
        return ViewHolder(itemListBinding)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        holder.itemListBinding.textViewChapterAll.text = allChapterDatasArrayList[position].chapterTitle
        holder.itemListBinding.linearViewChapterAll.setOnClickListener { v -> clickListener!!.onItemClick(position, v) }
    }

    override fun getItemCount(): Int {
        return allChapterDatasArrayList.size
    }

    inner class ViewHolder(val itemListBinding: ItemListSelectChapterBinding) : RecyclerView.ViewHolder(itemListBinding.root)

    interface ClickListener {
        fun onItemClick(position: Int, v: View)
    }
}
